src/util/DbrHelper.sol

```diff
@@ -0,0 +1,335 @@
+        if (dbrLeft > 0) dbr.transfer(params.toDbr, dbrLeft);
+        // Send leftover DOLA to the receiver
+        uint256 dolaLeft = dola.balanceOf(address(this));
+        if (dolaLeft > 0) dola.transfer(params.toDola, dolaLeft);
+    }
+
+    /// @notice Sell DBR amount for INV and deposit into the escrow
+    /// @param amount Amount of DBR to sell
+    /// @param minOutInv Minimum amount of INV to receive
+    /// @param to Address to receive INV deposit
+    /// @return invAmount Amount of INV deposited
+    function _sellAndDeposit(
+        uint256 amount,
+        uint256 minOutInv,
+        address to
+    ) internal returns (uint256 invAmount) {
+        // Sell DBR for INV
+        _sellDbr(amount, minOutInv, invIndex, address(this));
+        // Deposit INV
+        invAmount = inv.balanceOf(address(this));
+        invMarket.deposit(to, invAmount);
+
+        emit DepositInv(msg.sender, to, invAmount);
+    }
+
+    /// @notice Sell DBR amount for DOLA and repay debt
+    /// @param amount Amount of DBR to sell
+    /// @param minOutDola Minimum amount of DOLA to receive
+    /// @param repay Repay struct with parameters for repaying debt
+    /// @return dolaAmount Amount of DOLA received after selling DBR
+    /// @return repaidAmount Actual amount of DOLA repaid
+    function _sellAndRepay(
+        uint amount,
+        uint minOutDola,
+        Repay calldata repay
+    ) internal returns (uint256 dolaAmount, uint256 repaidAmount) {
+        // Sell DBR for DOLA
+        dolaAmount = _sellDbr(amount, minOutDola, dolaIndex, address(this));
+        // Repay debt
+        repaidAmount = _repay(repay, dolaAmount);
+    }
+
+    /// @notice Repay debt
+    /// @param repay Repay struct with parameters for repaying debt
+    /// @param dolaAmount Amount of DOLA available to repay
+    /// @return repaidAmount Actual amount of DOLA repaid
+    function _repay(
+        Repay calldata repay,
+        uint256 dolaAmount
+    ) internal returns (uint256 repaidAmount) {
+        uint256 debt = IMarket(repay.market).debts(repay.to);
+        repaidAmount = (dolaAmount * repay.percentage) / denominator;
+        // If repaidAmount is higher than debt, use debt instead
+        if (repaidAmount > debt) {
+            repaidAmount = debt;
+        }
+        // Repay debt
+        IMarket(repay.market).repay(repay.to, repaidAmount);
+
+        emit RepayDebt(msg.sender, repay.market, repay.to, repaidAmount);
+    }
+
+    /// @notice Claim DBR
+    /// @return amount of DBR claimed
+    function _claimDBR() internal returns (uint amount) {
+        IINVEscrow escrow = _getEscrow();
+
+        uint256 dbrClaimable = escrow.claimable();
+        if (dbrClaimable == 0) revert NoDbrToClaim();
+
+        escrow.claimDBRTo(address(this));
+        amount = dbr.balanceOf(address(this));
+        if (dbrClaimable > amount || amount == 0)
+            revert ClaimedWrongAmount(amount, dbrClaimable);
```

Comment:
Don't think it's necessary to do these checks when claiming DBR. Call only needs to fail if the trades are impossible. Or is there something I'm overlooking?
```suggestion
    function _claimDBR() internal returns (uint amount) {
        IINVEscrow escrow = _getEscrow();
        escrow.claimDBRTo(address(this));
        amount = dbr.balanceOf(address(this));
```

Reply:
Fixed in [f446755](https://github.com/InverseFinance/FiRM/pull/66/commits/f44675526e646aab388f30b936371beb701648ff)